### PR TITLE
Uar 495 add user submission data to new json template 24 may2023

### PR DESCRIPTION
**Pull request by vgummadala**
### JIRA link

https://companieshouse.atlassian.net/browse/UAR-495

### Change description

Replacement of https://github.com/companieshouse/overseas-entities-api/pull/243

### Work checklist

- [ ] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/PopulateUpdateSubmissionService.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+    public void populateDueDiligence(
+            OverseasEntitySubmissionDto overseasEntitySubmissionDto, UpdateSubmission updateSubmission) {
+
+        var dueDiligenceDto = overseasEntitySubmissionDto.getDueDiligence();
+        var overseasEntityDueDiligenceDto =
+                overseasEntitySubmissionDto.getOverseasEntityDueDiligence();
+        var submissionDueDiligence = new DueDiligence();
+        submissionDueDiligence =
+                (overseasEntitySubmissionDto.getDueDiligence() != null)
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/PopulateUpdateSubmissionService.java:42]: use var on line 37

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/PopulateUpdateSubmissionService.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                        ? populateByDueDiligenceDto(submissionDueDiligence, dueDiligenceDto)
+                        : populateByOEDueDiligenceDto(submissionDueDiligence, overseasEntityDueDiligenceDto);
+        updateSubmission.setDueDiligence(submissionDueDiligence);
+    }
+
+    public DueDiligence populateByDueDiligenceDto(
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/PopulateUpdateSubmissionService.java:48]: are these methods private? or package private if tests need that ...

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/PopulateUpdateSubmissionService.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        var filingDate = overseasEntitySubmissionDto.getUpdate().getFilingDate();
+        filingForDate.setDay((String.valueOf(filingDate.getDayOfMonth())));
+        filingForDate.setMonth((String.valueOf(filingDate.getMonth())));
+        filingForDate.setYear((String.valueOf(filingDate.getYear())));
+        updateSubmission.setFilingForDate(filingForDate);
+    }
+
+    public void populatePresenter(
+            OverseasEntitySubmissionDto overseasEntitySubmissionDto, UpdateSubmission updateSubmission) {
+        Presenter presenter = new Presenter();
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/PopulateUpdateSubmissionService.java:89]: again - no need for new

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/PopulateUpdateSubmissionService.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        dueDiligence.setSupervisoryBody(dueDiligenceDto.getSupervisoryName());
+        dueDiligence.setPartnerName(dueDiligenceDto.getPartnerName());
+        dueDiligence.setEmail(dueDiligenceDto.getEmail());
+        dueDiligence.setAgentAssuranceCode(dueDiligenceDto.getAgentCode());
+        dueDiligence.setDiligence(dueDiligenceDto.getDiligence());
+
+        return dueDiligence;
+    }
+
+    public DueDiligence populateByOEDueDiligenceDto(
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/PopulateUpdateSubmissionService.java:63]: no need to return anything as object passed in.

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/UpdateSubmissionTest.java [hunk 5]
```diff
@@ -13,28 +13,28 @@
...
+          + "\"presenter\":{\"name\":\"John Doe\",\"email\":\"john@test.com\"},"
+          + "\"filingForDate\":{\"year\":\"2000\",\"month\":\"01\",\"day\":\"01\"},"
+          + "\"noChangesInFilingPeriodStatement\":true,"
+          + "\"anyBOsOrMOsAddedOrCeased\":true,"
+          + "\"beneficialOwnerStatement\":\"BO Statement\","
+          + "\"changes\":[{\"change\":\"Change 1\"},{\"change\":\"Change 2\"}],"
+          + "\"additions\":[{\"change\":\"Addition 1\",\"type\":\"Type 1\"},{\"change\":\"Addition 2\",\"type\":\"Type 2\"}],"
+          + "\"cessations\":[{\"change\":\"Cessation 1\"},{\"change\":\"Cessation 2\"}]}";
+
     @Test
```
Comment (AndreKramer7654) [inline src/test/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/UpdateSubmissionTest.java:38]: remove this test

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/DueDiligence.java [hunk 6]
```diff
@@ -28,6 +28,9 @@ public class DueDiligence {
     @JsonProperty("partnerName")
```
[the remark this answers is not in the document]
Comment (bmcgreevyCH) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/DueDiligence.java:28]: Changed

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/PopulateUpdateSubmissionService.java [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+        if (overseasEntitySubmissionDto.getDueDiligence() != null) {
+            populateByDueDiligenceDto(submissionDueDiligence, dueDiligenceDto);
+        } else {
+            populateByOEDueDiligenceDto(submissionDueDiligence, overseasEntityDueDiligenceDto);
+        }
+        updateSubmission.setDueDiligence(submissionDueDiligence);
+    }
+
+    public void populateByDueDiligenceDto(
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/PopulateUpdateSubmissionService.java:49]: possibly this method and following don't need to be public.

**Review by AndreKramer7654 — APPROVED** [review]

**Review by bmcgreevyCH — APPROVED** [review]

**Pull request merged**